charms/istio-ingressgateway/src/charm.py

```diff
@@ -0,0 +1,229 @@
+                            {"name": "tracing", "containerPort": cfg['tracing-port']},
+                            {"name": "tls", "containerPort": cfg['tls-port']},
+                            {"name": "pilot", "containerPort": cfg['xds-ca-port-legacy']},
+                            {"name": "citadel", "containerPort": cfg['citadel-grpc-port']},
+                            {"name": "dns-tls", "containerPort": cfg['dns-tls-port']},
+                        ],
+                        "kubernetes": {
+                            "readinessProbe": {
+                                "failureThreshold": 30,
+                                "httpGet": {
+                                    "path": "/healthz/ready",
+                                    "port": cfg['status-port'],
+                                    "scheme": "HTTP",
+                                },
+                                "initialDelaySeconds": 1,
+                                "periodSeconds": 2,
+                                "successThreshold": 1,
+                                "timeoutSeconds": 1,
+                            },
+                        },
+                        "volumeConfig": [
+                            {
+                                "name": "istiod-ca-cert",
+                                "mountPath": "/var/run/secrets/istio",
+                                "files": [
+                                    {
+                                        "path": "root-cert.pem",
+                                        "content": config_map.data["root-cert.pem"],
+                                    }
+                                ],
+                            },
+                            {
+                                "name": "ingressgatewaysdsudspath",
+                                "mountPath": "/var/run/ingress_gateway",
+                                "emptyDir": {"medium": "Memory"},
+                            },
+                            {
+                                "name": "podinfo",
+                                "mountPath": "/etc/istio/pod",
+                                "files": [
+                                    {
+                                        "path": "annotations",
+                                        "content": 'sidecar.istio.io/inject="false"',
+                                    },
+                                    {
+                                        "path": "labels",
+                                        "content": 'app="istio-ingressgateway"\nistio="ingressgateway"',
+                                    },
+                                ],
+                            },
+                        ],
+                    }
+                ],
+            }
+        )
+
+        self.model.unit.status = ActiveStatus()
+
+    def check_ca_root_cert(self, namespace):
+        os.environ.update(
```

Comment:
Can this get a comment with a link to https://bugs.launchpad.net/juju/+bug/1892255? It seems likely to make future code readers that are out of the loop go wtf.